src/components/Popover/index.tsx

```diff
@@ -52,119 +36,105 @@ class Popover extends Component<PropsType, StateType> {
 
-        document.removeEventListener('mousedown', this.handleClickOutside, false);
-        document.removeEventListener('keydown', (event: KeyboardEvent) => this.handleKeyDown(event), false);
-    }
-
-    public render(): JSX.Element {
-        return (
-            <>
-                <Manager>
-                    <Reference>
-                        {({ ref }: ReferenceChildrenProps): JSX.Element => (
-                            <div
-                                ref={(ref): void => {
-                                    if (ref) this.anchorRef = ref;
-                                }}
-                            >
-                                <PopoverAnchor
-                                    ref={ref}
-                                    stretch={this.props.stretch}
-                                    role="button"
-                                    aria-expanded={this.state.isOpen}
-                                >
-                                    {this.props.children}
-                                </PopoverAnchor>
-                            </div>
+            return () => {
+                node.removeEventListener('mouseenter', e => handleMouseToggle(e, true));
+                node.removeEventListener('mouseleave', e => handleMouseToggle(e, false));
+            };
+        }
+    }, [anchorRef.current]);
+
+    useEffect(() => {
+        document.addEventListener(props.triggerOn === 'click' ? 'click' : 'mouseenter', handleToggleInside);
+        document.addEventListener('mousedown', handleClickOutside);
+        document.addEventListener('keydown', handleKeyDown);
+
+        return () => {
+            document.removeEventListener(props.triggerOn === 'click' ? 'click' : 'mouseenter', handleToggleInside);
+            document.removeEventListener('mousedown', handleClickOutside);
+            document.removeEventListener('keydown', handleKeyDown);
+        };
+    });
+
+    return (
+        <Manager>
+            <Reference>
+                {({ ref }: ReferenceChildrenProps): JSX.Element => (
+                    <div ref={anchorRef} style={{ display: 'table' }}>
+                        <PopoverAnchor ref={ref} stretch={props.stretch} role="button" aria-expanded={isOpen}>
+                            {props.children}
+                        </PopoverAnchor>
+                    </div>
+                )}
+            </Reference>
+            <TransitionAnimation show={props.show !== undefined ? props.show : isOpen} animation="fade">
+                <div ref={popoverRef}>
+                    <Popper
+                        positionFixed={!!props.fixed}
+                        placement={props.placement !== undefined ? props.placement : 'bottom'}
+                        modifiers={{
+                            offset: { offset: mapOffset(props) },
+                            flip: { enabled: false },
+                            preventOverflow: { enabled: !!props.preventOverflow },
```

Comment:
this now alters the behavior when undefined
```suggestion
preventOverflow: { enabled: props.preventOverflow !== undefined ? props.preventOverflow : true},
```